File: Stack.py
class Stack:
    def __init__(self):
        self.items = []

    def push(self, item):
        self.items.append(item)
    
    def pop(self, item):
        return self.items.pop()

    def clear(self):
        del self.items[:]

    def empty(self):
        return  len(self.items) == 0

    def size(self):
        return len(self.items)

    def top(self):
        return self.items[len(self.items) -1]

File: stack.go
package main

type Stack []byte

func (s *Stack) IsEmpy() bool {
	return len(*s) == 0
}

func (s *Stack) Push(c byte) {
	*s = append(*s, c)
}

func (s *Stack) Pop() (byte, bool) {
	if s.IsEmpy() {
		return 0, false
	}

	index := len(*s) - 1
	element := (*s)[index]
	*s = (*s)[:index]
	return element, true
}

func (s *Stack) Length() int {
	return len(*s)
}
